### PR TITLE
Remove Dead Code and Duplication from Metadata

### PR DESCRIPTION
**Pull request by original-brownbear**
Just some random spots I found while researching other things.

(only partly applies to `7.11` in a backport since the types utils need to be kept around there)

### CODE REVIEW

**Comment by original-brownbear** [comment]
@elasticmachine update branch

**Review by henningandersen — APPROVED** [review]
LGTM, thanks for doing this cleanup.

File: server/src/main/java/org/elasticsearch/action/admin/indices/get/TransportGetIndexAction.java [hunk 1]
```diff
@@ -82,13 +81,8 @@ protected void doMasterOperation(final GetIndexRequest request, String[] concret
...
             case MAPPINGS:
                     if (!doneMappings) {
-                        try {
-                            mappingsResult = state.metadata().findMappings(concreteIndices, indicesService.getFieldFilter());
-                            doneMappings = true;
-                        } catch (IOException e) {
-                            listener.onFailure(e);
-                            return;
-                        }
+                        mappingsResult = state.metadata().findMappings(concreteIndices, indicesService.getFieldFilter());
```
[the remark this answers is not in the document]
Comment (original-brownbear) [reply server/src/main/java/org/elasticsearch/action/admin/indices/get/TransportGetIndexAction.java:84]: I don't think I can remove those with the strange way these requests work (reading `features` from an array we could have duplicate entries)?

**Comment by original-brownbear** [comment]
Thanks Henning!

**Pull request merged**